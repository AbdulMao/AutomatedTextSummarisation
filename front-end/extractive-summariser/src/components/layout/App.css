.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 20s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  /* display: flex;
  flex-direction: column; */
  /* align-items: center; */
  /* justify-content: center; */
  font-size: calc(10px + 2vmin);
  color: white;
}

.Logo-container{
  text-align: center;
  height: 200px;
  background-image: url("../tldr.png");
  background-repeat: no-repeat;
  background-position: center; 
  filter: invert(100%);
}

.Info-container{
  text-align: center;
}

.Input-container{
  text-align: center;
  height: 383px;
}

.Submit-container{
  text-align: center;

}

.textbox{
  margin-right: 75px;
  margin-top: 65px;
}

.Results-page{
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.Summary-box{
    background: white;
    color: black;
    font-size: 15px;
    width: 600px;
    height: 500px;
    overflow: scroll;
    text-align: left;
}

.App-link {
  color: #61dafb;
}

.Submit-button{
  color: #fff !important;
  text-transform: uppercase;
  background: #ed3330;
  padding: 12px;
  border-radius: 5px;
  display: inline-block;
  border: none;
  margin-right: 75px;
  }

.Submit-button:hover{
  background: rgb(225, 0, 80);
  letter-spacing: 1px;
  -webkit-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  -moz-box-shadow: 0px 5px 40px -10px rgba(0,0,0,0.57);
  box-shadow: 5px 40px -10px rgba(0,0,0,0.57);
  transition: all 0.4s ease 0s;
  }

@media all and (max-width:30em){
.Submit-button{
  display:block;
  margin:0.4em auto;
  }

  }

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}
