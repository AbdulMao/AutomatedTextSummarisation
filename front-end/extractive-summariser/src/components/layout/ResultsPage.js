import React, { Component } from 'react';
import makeSummaryRequest from '../submit/SummaryRequest.js';
import makeTopicRequest from '../submit/TopicRequest.js';
import SummaryBox from '../results/SummaryBox.js'
import TopicBox from '../results/TopicBox.js'


class ResultsPage extends Component {
    constructor(){
      super()  
      this.state = {}
    }

    componentDidMount(){    
        const inputText = this.props.location.state.myText;
        var myJSON = {"text": inputText };
        makeTopicRequest(myJSON).then( (dataReturned) => {
            this.setState({
               myTopics: dataReturned.topics, 
            })
        })
        makeSummaryRequest(myJSON).then( (dataReturned) => {
            this.setState({
               mySummary: dataReturned, 
            })
        })
    

    }


    render(){
        if(!this.state.myTopics || !this.state.mySummary) return <p> loading </p>
        return (
            <div className="App">              
                <body className="App-header">
                    <div className="Logo-container"></div>
                    <div className="Info-container">Here is your Summary:</div>
                        <SummaryBox summary= {this.state.mySummary} percentage= {this.props.location.state.myPercentage} />
                        <TopicBox topics= {this.state.myTopics} />    
                </body>
            </div>   
              
        );
    }
    
}

export default ResultsPage;