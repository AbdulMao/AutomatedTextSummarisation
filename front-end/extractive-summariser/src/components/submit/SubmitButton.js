import React, { Component } from 'react';
import { Link, withRouter} from "react-router-dom"
import makeSummaryRequest from './SummaryRequest.js';
import makeTopicRequest from './TopicRequest.js';
import PropTypes from 'prop-types';
import { hashHistory } from 'react-router-dom';


class SubmitButton extends Component {
  constructor(){
    super()  
  }
  onClick = () => {
    console.log("WE GOT THIS FAR")
    const mytext = document.querySelector(".textbox").value;
    console.log(mytext)
    this.props.history.push({
        pathname: '/summary',
        state: {
            myText: mytext
        }            
    })
    // const text = document.querySelector(".textbox").value;
    // var myJSON = {"text": text };
    // makeSummaryRequest(myJSON);
    // makeTopicRequest(myJSON);
    // console.log(myJSON)
    
}

  render(){
      return (
          <button onClick={this.onClick} className ="Submit-button" >
            Submit
          </button> 
      );
  }
}

export default withRouter(SubmitButton);