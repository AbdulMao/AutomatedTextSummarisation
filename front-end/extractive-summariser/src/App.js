import React, { Component } from 'react';
import logo from './logo.svg';
import './App.css';
import TextBox from './TextBox.js'
import SubmitButton from './SubmitButton.js'


class App extends Component {
  render() {
    return (
      <div className="App">
        <body className="App-header">
          {/* <img src={logo} className="App-logo" alt="logo" /> */}
          <TextBox />
          <SubmitButton />
        </body>
      </div>
    );
  }
}

export default App;
