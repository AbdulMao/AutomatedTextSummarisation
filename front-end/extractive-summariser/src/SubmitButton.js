import React, { Component } from 'react';
import './App.css';
import PropTypes from 'prop-types';
import { withStyles } from '@material-ui/core/styles';
import Button from '@material-ui/core/Button';


const styles = theme => ({
  button: {
    margin: theme.spacing.unit,
  },
  input: {
    display: 'none',
  },
});

function dothis(){
  const text = document.querySelector(".textbox").value;
  console.log(text)
}

function SubmitButton(props){
  const { classes } = props;
    return (
      <Button onClick={dothis} variant="contained" color="secondary" className={classes.button} id = "submitButton">
        Submit
      </Button>
    );
}

SubmitButton.propTypes = {
  classes: PropTypes.object.isRequired,
};

export default withStyles(styles)(SubmitButton);
