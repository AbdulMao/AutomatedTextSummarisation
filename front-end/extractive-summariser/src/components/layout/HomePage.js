import React, { Component } from 'react';
import TextBox from '../textBox/TextBox.js' 
import SubmitButton from '../submit/SubmitButton.js'
import makeSummaryRequest from '../submit/SummaryRequest.js';
import makeTopicRequest from '../submit/TopicRequest.js';


class HomePage extends Component {
    constructor(props){
      super(props)  
    }

    render(){
        return (
            <div className="App">
                <body className="App-header">
                    <div className="Logo-container"></div>
                    <div className="Info-container">INFO</div>
                    <div className="Input-container">
                        <TextBox />
                        {/* <select>
                            <option value="volvo">50%</option>
                            <option value="saab">30%</option>
                            <option value="opel">20%</option>
                            <option value="audi">10%</option>
                        </select> */}
                    </div>
                    <div className="Submit-container">
                        <SubmitButton/>
                    </div>                        
                </body>
            </div>        
        );
    }
}

export default HomePage;