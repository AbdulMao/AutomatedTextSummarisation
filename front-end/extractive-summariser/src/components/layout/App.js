import React, { Component } from 'react';
import { BrowserRouter, Route, Switch } from "react-router-dom"
// import logo from './logo.svg';
import './App.css';
import TextBox from '../textBox/TextBox.js' 
import SubmitButton from '../submit/SubmitButton.js'
import HomePage from './HomePage.js'
import ResultsPage from './ResultsPage.js'


class App extends Component {
  render() {
    return (
      <BrowserRouter>
        <Switch>
          <Route path="/" component={HomePage} exact />
          <Route path="/summary" component={ResultsPage}  />
        </Switch>   
      </BrowserRouter>
    );
  }
}

export default App;
