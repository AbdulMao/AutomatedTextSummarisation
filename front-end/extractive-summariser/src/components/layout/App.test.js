import React from 'react';
import ReactDOM from 'react-dom';
import App from './App';
import HomePage from './HomePage.js'
import renderer from 'react-test-renderer'
import { MemoryRouter as Router, withRouter } from 'react-router-dom';
import ResultsPage from './ResultsPage.js';
import SubmitButton from '../results/SummaryBox.js'

const returnPercentageList = jest.fn(() => 4);
const homepagePageInfo = jest.fn(() => "Too Long; Didn't Read.Quickly get the information you need, saving time! ");
const topics = jest.fn(() => ["Mars", "Pluto", "Uranus", "Earth"])
const summary = jest.fn(()=> "This is the summary created")

it('renders without crashing', () => {
  const div = document.createElement('div');
  ReactDOM.render(<App />, div);
  ReactDOM.unmountComponentAtNode(div);
});

it('Homepage matches snapshot', () => {
  const tree = renderer.create(<Router
    initialEntries={[ { pathname: '/'} ]}
  >
      <HomePage/>
     </Router>).toJSON()
  expect(tree).toMatchSnapshot()
});

test ('all percentage choices are presented', ()=> {
  expect(returnPercentageList()).toBe(4);
})

test ('Home page information is correct', ()=> {
  expect(homepagePageInfo()).toBe("Too Long; Didn't Read.Quickly get the information you need, saving time! ");
})

test ('Topics list should contain 4 topcics', ()=> {
  expect(topics()).toHaveLength(4);
})

test ('Summary Exists', ()=> {
  expect(summary()).toBeTruthy();
})

