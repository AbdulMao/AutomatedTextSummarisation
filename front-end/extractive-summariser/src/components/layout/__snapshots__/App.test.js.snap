// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Homepage matches snapshot 1`] = `
<div
  className="App"
>
  <body
    className="App-header"
  >
    <div
      className="Logo-container"
    />
    <div
      className="Info-container"
    >
      Too Long; Didn't Read. Quickly get the information you need, saving time!
    </div>
    <div
      className="Percentage-holder"
      id="Percentage-holder"
    >
      Choose summary size: Â 
      <select
        className="Percentage-values"
      >
        <option
          value="50"
        >
          50%
        </option>
        <option
          value="30"
        >
          30%
        </option>
        <option
          value="20"
        >
          20%
        </option>
        <option
          value="10"
        >
          10%
        </option>
      </select>
    </div>
    <div
      className="Input-container"
    >
      <div
        className="textbox-holder"
      >
        <textarea
          className="textbox"
          cols="70"
          placeholder="Enter text to be summarised here"
          rows="20"
        />
      </div>
    </div>
    <div
      className="Submit-container"
    >
      <button
        className="Submit-button"
        onClick={[Function]}
      >
        Submit
      </button>
    </div>
  </body>
</div>
`;
